### PR TITLE
adds dependpendency smoketest

### PR DESCRIPTION
**Pull request by ededals**
Adds dependency smoketest for #1242

### CODE REVIEW

**Comment by djmitche** [comment]
If you rebase onto the latest master, you'll get the full set of CI tests.

**Comment by ededals** [comment]
Is this it?

**Comment by djmitche** [comment]
Much better :)

File: infrastructure/tooling/src/smoketest/checks/dependencies.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+const taskcluster = require('taskcluster-client');
+exports.tasks = [];
+exports.tasks.push({
+  title: 'Check dependencies',
+  requires: [],
+  provides: [
+    'dependency smoketest',
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/dependencies.js]:
```suggestion
    'target-dependencies',
```

This means we can run `yarn smoketest --target dependencies` to run just this check

File: infrastructure/tooling/src/smoketest/checks/dependencies.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        deadline: taskcluster.fromNowJSON('2 minutes'),
+        metadata: {
+          name: 'Smoketest dependencies task Nr' + (3-i),
+          description: 'built-in/succeed task created during dependency smoketest',
+          owner: 'smoketest@taskcluster.net',
+          source: 'https://taskcluster.net',
+        },
+        payload: {},
+      };
+      task.dependencies =[...taskIds];
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/dependencies.js]:
```suggestion
      task.dependencies = [...taskIds];
```

File: infrastructure/tooling/src/smoketest/checks/dependencies.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    let taskIds = [];
+    let queue = new taskcluster.Queue(taskcluster.fromEnvVars());
+    for (let i = 0; i < taskCount; i++){
+      let task ={
+        provisionerId: 'built-in',
+        workerType: 'succeed',
+        created: (new Date()).toJSON(),
+        deadline: taskcluster.fromNowJSON('2 minutes'),
+        metadata: {
+          name: 'Smoketest dependencies task Nr' + (3-i),
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/dependencies.js]:
```suggestion
          name: 'Smoketest dependencies task Nr' + (taskCount-i),
```

File: infrastructure/tooling/src/smoketest/checks/dependencies.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            }
+
+          }
+        }
+        statuses.push(taskStatus);
+      }
+      utils.status({message: message});
+      if (statuses[statuses.length-1].status.state === 'completed') {
+        return {
+          ['dependency smoketest']: statuses[statuses.length-1].status.state,
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/dependencies.js]: change this to `target-dependencies`, too

**Comment by ededals** [comment]
I made the corrections. Should I try to why two of the tests are not passing, or is it ok?

**Pull request merged**